test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -954,4 +1003,60 @@ def nothing() {
 }
 nothing();"""
 
-    assert main().to_ir() == expected
+    assert main.to_ir() == expected
+
+
+def test_input_types():
```

Comment:
The title of this test had me looking at the types of x, y and z.
```suggestion
def test_main_decorator_args():
```

Reply:
Yes, that's correct. The test asserts that x, y, z, u get captured as inputs of the correct types. It does this for multiple decorator setups, but the main focus is the input types. I think the test name is valid

Reply:
That's very confusing